src/test/java/seedu/address/logic/commands/AddAltCommandTest.java

```diff
@@ -0,0 +1,159 @@
+                null, new AddAltCommand.AddAltPersonDescriptor())
+        );
+        assertThrows(NullPointerException.class, () -> new AddAltCommand(null, null));
+    }
+
+    @Test
+    public void execute_personEditsModel_setSuccessful() throws Exception {
+        Person personA = new PersonBuilder().withName("A").build();
+
+        AddAltCommand.AddAltPersonDescriptor descriptor = new AddAltCommand.AddAltPersonDescriptor();
+        descriptor.setLinkedin(new Linkedin("LINKEDIN"));
+        descriptor.setSecondaryEmail(new Email("email@email.com"));
+        descriptor.setTelegram(new Telegram("@telegram"));
+
+        ModelStubSetPerson modelStub = new ModelStubSetPerson(personA, p -> {
+            assertEquals(p.getName(), new Name("A"));
+            assertEquals(p.getLinkedin().orElseThrow(), new Linkedin("LINKEDIN"));
+            assertEquals(p.getSecondaryEmail().orElseThrow(), new Email("email@email.com"));
+            assertEquals(p.getTelegram().orElseThrow(), new Telegram("@telegram"));
+        });
+
+        CommandResult commandResult = new AddAltCommand(Index.fromZeroBased(0), descriptor).execute(modelStub);
+    }
+
+    @Test
+    public void addAltPersonDescriptor() {
+        AddAltCommand.AddAltPersonDescriptor descriptor = new AddAltCommand.AddAltPersonDescriptor();
+        descriptor.setTelegram(new Telegram("@telegram"));
+        assertEquals(descriptor.getTelegram(), new Telegram("@telegram"));
+        descriptor.setLinkedin(new Linkedin("LINKEDIN"));
+        assertEquals(descriptor.getLinkedin(), new Linkedin("LINKEDIN"));
+        descriptor.setSecondaryEmail(new Email("email@email.com"));
+        assertEquals(descriptor.getSecondaryEmail(), new Email("email@email.com"));
+    }
+
+    @Test
+    public void equals() {
+        AddAltCommand.AddAltPersonDescriptor descriptorA = new AddAltCommand.AddAltPersonDescriptor();
+        AddAltCommand.AddAltPersonDescriptor descriptorB = new AddAltCommand.AddAltPersonDescriptor();
+        descriptorA.setTelegram(new Telegram("@atelegram"));
+        descriptorB.setTelegram(new Telegram("@btelegram"));
+        AddAltCommand commandA = new AddAltCommand(Index.fromZeroBased(0), descriptorA);
+        AddAltCommand commandB = new AddAltCommand(Index.fromZeroBased(1), descriptorB);
+
+        // same object -> returns true
+        assertTrue(commandA.equals(commandA));
+
+        // same values -> returns true
+        AddAltCommand commandACopy = new AddAltCommand(Index.fromZeroBased(0), descriptorA);
+        assertTrue(commandA.equals(commandACopy));
+
+        // different types -> returns false
+        assertFalse(commandA.equals(1));
+
+        // null -> returns false
+        assertFalse(commandA.equals(null));
+
+        // different args -> returns false
+        assertFalse(commandA.equals(commandB));
+    }
+
+    /**
+     * A Model stub that contains a single person.
+     */
+    private class ModelStubSetPerson extends ModelStub {
```

Comment:
If we're going to be testing more methods using this, would suggest taking out the ModelStub into it's own class rather than leaving inside AddCommandTest.java.

Reply:
Have made it into issue #70